File: src/main/java/com/regionrental/managers/WorldGuardManager.java
package com.regionrental.managers;

import com.regionrental.RegionRental;
import com.sk89q.worldedit.bukkit.BukkitAdapter;
import com.sk89q.worldguard.LocalPlayer;
import com.sk89q.worldguard.WorldGuard;
import com.sk89q.worldguard.bukkit.WorldGuardPlugin;
import com.sk89q.worldguard.domains.DefaultDomain;
import com.sk89q.worldguard.protection.managers.RegionManager;
import com.sk89q.worldguard.protection.regions.ProtectedRegion;
import org.bukkit.Bukkit;
import org.bukkit.World;
import org.bukkit.entity.Player;

import java.util.UUID;

public class WorldGuardManager {
    
    private final RegionRental plugin;
    private WorldGuardPlugin worldGuard;
    
    public WorldGuardManager(RegionRental plugin) {
        this.plugin = plugin;
        setupWorldGuard();
    }
    
    private void setupWorldGuard() {
        org.bukkit.plugin.Plugin wgPlugin = plugin.getServer().getPluginManager().getPlugin("WorldGuard");
        
        if (wgPlugin == null || !(wgPlugin instanceof WorldGuardPlugin)) {
            plugin.getLogger().severe("WorldGuard not found!");
            return;
        }
        
        this.worldGuard = (WorldGuardPlugin) wgPlugin;
        plugin.getLogger().info("WorldGuard integration enabled");
    }
    
    public boolean regionExists(String regionName, World world) {
        if (worldGuard == null) return false;
        
        RegionManager regionManager = WorldGuard.getInstance().getPlatform()
                .getRegionContainer().get(BukkitAdapter.adapt(world));
        
        if (regionManager == null) return false;
        
        return regionManager.hasRegion(regionName);
    }
    public void addPlayerToRegion(String regionName, World world, UUID playerUUID) {
        RegionManager regionManager = WorldGuard.getInstance().getPlatform()
                .getRegionContainer().get(BukkitAdapter.adapt(world));

        if (regionManager == null) {
            plugin.getLogger().warning("No RegionManager found for world: " + world.getName());
            return;
        }

        ProtectedRegion region = regionManager.getRegion(regionName);
        if (region != null) {
            DefaultDomain members = region.getMembers();
            members.addPlayer(playerUUID);

            plugin.getLogger().info("Added player " + playerUUID + " to region " + regionName + " in world " + world.getName());
        } else {
            plugin.getLogger().warning("Region " + regionName + " not found in world " + world.getName());
        }
    }
    public void removePlayerFromRegion(String regionName, World world, UUID playerUUID) {
        RegionManager regionManager = WorldGuard.getInstance().getPlatform()
                .getRegionContainer().get(BukkitAdapter.adapt(world));

        if (regionManager == null) {
            plugin.getLogger().warning("No RegionManager found for world: " + world.getName());
            return;
        }

        ProtectedRegion region = regionManager.getRegion(regionName);
        if (region != null) {
            DefaultDomain members = region.getMembers();
            members.removePlayer(playerUUID);

            plugin.getLogger().info("Removed player " + playerUUID + " from region " + regionName + " in world " + world.getName());
        } else {
            plugin.getLogger().warning("Region " + regionName + " not found in world " + world.getName());
        }
    }
    public ProtectedRegion getRegion(String regionName, World world) {
        if (worldGuard == null) return null;
        
        RegionManager regionManager = WorldGuard.getInstance().getPlatform()
                .getRegionContainer().get(BukkitAdapter.adapt(world));
        
        if (regionManager == null) return null;
        
        return regionManager.getRegion(regionName);
    }

    /**
     * Get all region names across all worlds for tab completion
     */
    public java.util.Set<String> getAllRegionNames() {
        java.util.Set<String> regionNames = new java.util.HashSet<>();

        for (World world : Bukkit.getWorlds()) {
            RegionManager regionManager = WorldGuard.getInstance().getPlatform()
                    .getRegionContainer().get(BukkitAdapter.adapt(world));

            if (regionManager == null) continue;

            regionNames.addAll(regionManager.getRegions().keySet());
        }

        return regionNames;
    }
}

File: src/main/kotlin/com/regionrental/managers/WorldGuardManager.kt
package com.regionrental.managers

import com.regionrental.RegionRental
import com.sk89q.worldedit.bukkit.BukkitAdapter
import com.sk89q.worldguard.WorldGuard
import com.sk89q.worldguard.bukkit.WorldGuardPlugin
import com.sk89q.worldguard.protection.regions.ProtectedRegion
import org.bukkit.Bukkit
import org.bukkit.World
import java.util.UUID

/**
 * Manages WorldGuard region operations for the rental system.
 */
class WorldGuardManager(private val plugin: RegionRental) {

    private var worldGuard: WorldGuardPlugin? = null

    init {
        setupWorldGuard()
    }

    private fun setupWorldGuard() {
        val wgPlugin = plugin.server.pluginManager.getPlugin("WorldGuard")

        if (wgPlugin == null || wgPlugin !is WorldGuardPlugin) {
            plugin.logger.severe("WorldGuard not found!")
            return
        }

        worldGuard = wgPlugin
        plugin.logger.info("WorldGuard integration enabled")
    }

    fun regionExists(regionName: String, world: World): Boolean {
        if (worldGuard == null) return false

        val regionManager = WorldGuard.getInstance().platform
            .regionContainer.get(BukkitAdapter.adapt(world))
            ?: return false

        return regionManager.hasRegion(regionName)
    }

    fun addPlayerToRegion(regionName: String, world: World, playerUUID: UUID) {
        val regionManager = WorldGuard.getInstance().platform
            .regionContainer.get(BukkitAdapter.adapt(world))

        if (regionManager == null) {
            plugin.logger.warning("No RegionManager found for world: ${world.name}")
            return
        }

        val region = regionManager.getRegion(regionName)
        if (region != null) {
            region.members.addPlayer(playerUUID)
            plugin.logger.info("Added player $playerUUID to region $regionName in world ${world.name}")
        } else {
            plugin.logger.warning("Region $regionName not found in world ${world.name}")
        }
    }

    fun removePlayerFromRegion(regionName: String, world: World, playerUUID: UUID) {
        val regionManager = WorldGuard.getInstance().platform
            .regionContainer.get(BukkitAdapter.adapt(world))

        if (regionManager == null) {
            plugin.logger.warning("No RegionManager found for world: ${world.name}")
            return
        }

        val region = regionManager.getRegion(regionName)
        if (region != null) {
            region.members.removePlayer(playerUUID)
            plugin.logger.info("Removed player $playerUUID from region $regionName in world ${world.name}")
        } else {
            plugin.logger.warning("Region $regionName not found in world ${world.name}")
        }
    }

    fun getRegion(regionName: String, world: World): ProtectedRegion? {
        if (worldGuard == null) return null

        val regionManager = WorldGuard.getInstance().platform
            .regionContainer.get(BukkitAdapter.adapt(world))
            ?: return null

        return regionManager.getRegion(regionName)
    }

    /**
     * Get all region names across all worlds for tab completion.
     */
    fun getAllRegionNames(): Set<String> {
        val regionNames = mutableSetOf<String>()

        for (world in Bukkit.getWorlds()) {
            val regionManager = WorldGuard.getInstance().platform
                .regionContainer.get(BukkitAdapter.adapt(world))
                ?: continue

            regionNames.addAll(regionManager.regions.keys)
        }

        return regionNames
    }
}
